packages/arb-avm-cpp/avm_values/src/tuple.cpp

```diff
@@ -114,47 +105,79 @@ Tuple::Tuple(value val1,
     tpl->data.push_back(std::move(val8));
 }
 
-Tuple::Tuple(std::vector<value> values, TuplePool* pool) : tuplePool(pool) {
+Tuple::Tuple(std::vector<value> values) {
     if (!values.empty() && values.size() < 9) {
-        tpl = pool->getResource(values.size());
+        tpl = TuplePool::get_impl().getResource(values.size());
         for (auto& val : values) {
             tpl->data.push_back(std::move(val));
         }
     }
 }
 
-void Tuple::marshal(std::vector<unsigned char>& buf) const {
-    buf.push_back(TUPLE + tuple_size());
-    for (uint64_t i = 0; i < tuple_size(); i++) {
-        marshal_value(get_element(i), buf);
+constexpr uint64_t hash_size = 32;
+
+struct BasicValChecker {
```

Comment:
Leave a comment explaining what property of the object the "ValChecker" is checking.

Reply:
Added a comment